### PR TITLE
Issue #8 - Add support for Android projects

### PR DESCRIPTION
**Pull request by Tapchicoma**
Fix for issue #8: This pull request adds support for Android projects with applied `kotlin-android` plugin. Also adds applying plugin only to root project - it will scan subprojects and add tasks accordingly.

### CODE REVIEW

**Comment by JLLeitschuh** [comment]
I'd still like to be able to run `:[someProject]:ktlintCheck` to only validate one project.
I work on projects with a lot of subprojects and there are sometimes cases where you don't care about other projects during your development cycle.

**Comment by JLLeitschuh** [comment]
Thanks for the PR! I'll look over it a bit more in a bit when I have a free moment.

**Comment by Tapchicoma** [comment]
You can still apply plugin only for subproject. Applying to root project is optional.

**Comment by JLLeitschuh** [comment]
Also, would you be so kind as to update the change log with your change listed under the header.
```
## [Unreleased]
```

I will update the header when I do a release.

File: src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintPlugin.kt [hunk 1]
```diff
@@ -1,8 +1,11 @@
 package org.jlleitschuh.gradle.ktlint
 
+import com.android.build.gradle.BaseExtension
```
Comment (Tapchicoma) [inline src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintPlugin.kt]: Checked on pure Kotlin project (without any android dependencies) - works ok. When I remove this `compileOnly` dependency - plugin isn't compiled.

File: README.md [hunk 2]
```diff
@@ -61,7 +61,8 @@ ktlint {
 
 This plugin adds two tasks to every source set: `ktlint[source set name]Check` and `ktlint[source set name]Format`.
 Additionally, a simple `ktlintCheck` task has also been added that checks all of the source sets for that project.
-Similarly, a `ktlintFormat` task has been added that formats all of the source sets.
+Similarly, a `ktlintFormat` task has been added that formats all of the source sets. Also same meta tasks are added
+to the root project for convenience - they will call related tasks in subprojects.
```
Comment (Tapchicoma) [inline README.md]: True that user can see tasks from subprojects as well.

Previously when user applied plugin to the root project that contains subprojects - no ktlint tasks are added, but now user can do that - plugin will add tasks for subprojects and add them as a dependency in the root project meta task.

Will make description more understandable.

Comment (JLLeitschuh) [inline README.md]: Thanks! This is the first plugin I've published so it's very likely I did something goofy.

File: src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintPlugin.kt (hunk 1 above)
Comment (JLLeitschuh) [inline src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintPlugin.kt]: Okay, thanks for checking!

File: README.md (hunk 2 above)
Comment (Tapchicoma) [inline README.md]: Fixed in 32db797

**Comment by Tapchicoma** [comment]
Updated CHANGELOG in 704ab7b

**Pull request merged**